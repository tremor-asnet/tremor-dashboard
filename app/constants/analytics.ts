/* Constants for Analytics Page */
export const SALE_STATISTICAL_TYPE = {
  REVENUE: "Revenue",
};
