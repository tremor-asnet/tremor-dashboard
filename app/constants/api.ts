import { ROUTER_API_URL } from "./routes";

export const API_ROUTES = {
  USERS: `${ROUTER_API_URL}/users`,
};
