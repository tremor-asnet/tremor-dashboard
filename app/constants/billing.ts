export const enum AGGREGATION_TYPE {
  SALARY = "Salary",
  PAYPAL = "Paypal",
}

export const enum AGGREGATION_DESCRIPTION {
  SALARY = "Belong Interactive",
  PAYPAL = "Freelance Payment",
}
