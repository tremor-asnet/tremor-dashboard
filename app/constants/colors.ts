type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const CHANNELS_CHART_COLOR: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};

export const TRANSACTION_COLOR = {
  INCREASE: "few",
  DECREASE: "attention",
  PENDING: "secondary",
};
