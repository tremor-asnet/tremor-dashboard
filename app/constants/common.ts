export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export const DIRECTION = {
  ASC: "asc",
  DESC: "desc",
};
