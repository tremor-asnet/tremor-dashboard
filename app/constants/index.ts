export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./images";
export * from "./sections";
export * from "./images";
export * from "./products";
export * from "./colors";
export * from "./units";
export * from "./sales";
export * from "./commons";
export * from "./status";
