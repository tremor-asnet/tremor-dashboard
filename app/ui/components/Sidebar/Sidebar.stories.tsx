import type { Meta, StoryObj } from "@storybook/react";

// Components
import SideBar from "./Sidebar";

// Constants
import { ROUTES } from "@/constants";

const meta = {
  title: "Components/Common/SideBar",
  component: SideBar,
  tags: ["autodocs"],
  argTypes: {
    avatarUrl: { description: "Src of user avatar on side bar" },
    name: { description: "User name on side bar" },
    pathname: { description: "Path name of page on side bar" },
    toggleSidebar: { description: "Function toogle icon of side bar" },
    isCollapse: { description: "Set state collapse for side bar" },
    onSignOut: { description: "Function logout user on side bar" },
  },
} as Meta<typeof SideBar>;

export default meta;

type Story = StoryObj<typeof meta>;

export const SideBarDashboard: Story = {
  args: {
    avatarUrl: "/images/avatar/avatar-sm.webp",
    name: "Brooklyn Alice",
    pathname: ROUTES.PROJECTS,
    toggleSidebar: () => {},
    isCollapse: true,
    onSignOut: function (): Promise<void> {
      throw new Error("Function not implemented.");
    },
  },
};
