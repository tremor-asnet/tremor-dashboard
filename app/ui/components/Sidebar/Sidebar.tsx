"use client";

import { useEffect, useRef } from "react";

// lib
import Link from "next/link";
import { FaAngleLeft, FaAngleRight } from "react-icons/fa";

import {
  Accordion,
  AccordionBody,
  AccordionHeader,
  AccordionList,
  Flex,
  Metric,
  List,
  ListItem,
  Text,
} from "@tremor/react";

// Components
import { Avatar, CustomImage } from "@/ui/components";

// Constants
import { ITEMS_PROFILE, ITEMS_DASHBOARD, ROUTES, LOGO_SRC } from "@/constants";
import { BREAKPOINTS } from "@/constants/breakpoints";

// Styles
import "./styles.css";

// Helpers
import { isBrowser } from "@/helpers";

interface SideBarProps {
  avatarUrl: string;
  name: string;
  pathname: string;
  isCollapse: boolean;
  toggleSidebar: () => void;
  onSignOut: () => Promise<void>;
}

const SideBar = ({
  avatarUrl,
  name,
  isCollapse,
  toggleSidebar,
  pathname,
  onSignOut,
}: SideBarProps) => {
  const sideBarRef = useRef<HTMLDivElement>(null);
  const hiddenOpenClass = isCollapse && "xl:hidden";
  const centerOpenClass = isCollapse && "xl:justify-center";
  const transitionBgClass =
    "transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20";

  useEffect(() => {
    const handleClickOutside = (event: Event) => {
      const currentRef = sideBarRef.current;

      if (currentRef && !currentRef.contains(event.target as Node)) {
        // Clicked outside the side navigation bar, close it
        if (isCollapse) {
          toggleSidebar();
        }
      }
    };

    // Add event listener to the document object
    document.addEventListener("mousedown", handleClickOutside);

    // Remove event listener when the component unmounts
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [isCollapse, toggleSidebar]);

  // Handle case close sidebar in smaller than a desktop screen
  const handleClickSidebarItem = () => {
    if (isBrowser && window.innerWidth <= BREAKPOINTS.XL) {
      toggleSidebar();
    }
  };

  return (
    <>
      <div
        ref={sideBarRef}
        className={`sidebar antialiased shadow-box-sidebar bg-gradient-primary dark:bg-none dark:bg-dark-gradient-primary w-[250px] rounded-xl z-50 px-4 pt-6 overflow-y-auto overflow-x-hidden fixed top-4 left-4 h-[calc(100vh-2rem)] transition-all ease-in ${
          isCollapse
            ? "translate-x-0 xl:w-[100px] delay-10 duration-300"
            : "translate-x-[-20rem] xl:translate-x-0 xl:w-[260px] delay-10 duration-300"
        }`}>
        <Flex justifyContent="end">
          <div
            className="border p-1 rounded-s-md -mr-4 bg-white cursor-pointer"
            onClick={toggleSidebar}>
            {isCollapse ? <FaAngleLeft /> : <FaAngleRight />}
          </div>
        </Flex>
        <Link href={ROUTES.HOME}>
          <Flex className="justify-normal pl-5 gap-1 pb-2 flex-nowrap relative">
            <CustomImage
              src={LOGO_SRC.logo}
              width={28}
              height={28}
              alt="Tremor Dashboard"
            />
            <Metric
              className={`text-white text-tremor-default min-w-[200px] flex flex-nowrap ${hiddenOpenClass} delay-300 duration-500`}>
              Tremor Dashboard PRO
            </Metric>
          </Flex>
        </Link>
        <div className="h-px bg-gradient-divider my-4 opacity-25" />
        <AccordionList className="bg-transparent shadow-none border-0">
          <Accordion className="bg-inherit dark:bg-none dark:bg-dark-gradient-primary border-0 rounded-md">
            <AccordionHeader
              className={`min-w-[210px] flex text-[rgba(255,255,255,0.5)] py-2 ${
                isCollapse && "xl:w-[200%]"
              }`}>
              <Avatar
                alt="Avatar small"
                className="flex items-center justify-center"
                width={36}
                height={36}
                priority
                src={avatarUrl}
              />
              <Text
                className={`leading-0 dark:text-dark-primary min-w-[100px] flex flex-nowrap text-white self-center ml-[7px] ${hiddenOpenClass}`}>
                {name}
              </Text>
            </AccordionHeader>
            <AccordionBody>
              <List>
                {ITEMS_PROFILE.map(item => {
                  const { label, href, content } = item;
                  const isActive = href === pathname;
                  const itemBackground = isActive
                    ? "bg-primary hover:!bg-primary"
                    : "hover:bg-none";

                  const menuItemClass = [
                    `${itemBackground} `,
                    `!p-0 leading-[26px] mt-1 rounded-md text-center`,
                    `${transitionBgClass}`,
                  ].join("");

                  return (
                    <ListItem
                      className={menuItemClass}
                      key={label}
                      onClick={handleClickSidebarItem}>
                      <Link href={href}>
                        <Flex
                          className={`w-full gap-6 font-normal py-3 px-7 ${centerOpenClass}`}>
                          <span className="w-5">{content}</span>
                          <span
                            className={`${hiddenOpenClass} w-full text-start`}>
                            {label}
                          </span>
                        </Flex>
                      </Link>
                    </ListItem>
                  );
                })}
                <ListItem className="leading-[26px] relative !p-0 mt-1">
                  <Flex
                    className={`w-full gap-6 font-normal py-3 px-7 ${centerOpenClass}`}>
                    <span className="w-5">L</span>
                    <button
                      onClick={onSignOut}
                      className={`${hiddenOpenClass} w-full text-start`}>
                      Logout
                    </button>
                  </Flex>
                </ListItem>
              </List>
            </AccordionBody>
          </Accordion>
        </AccordionList>
        <div className="h-px bg-gradient-divider my-4 opacity-25" />
        <ul>
          {ITEMS_DASHBOARD.map(({ label, content, href }) => {
            const isActive = href === pathname;
            const itemBackground = isActive
              ? "bg-primary hover:!bg-primary"
              : "hover:bg-none";

            const menuItemClass = [
              `tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center `,
              `${itemBackground} `,
              `!p-0 leading-[26px] mt-1 rounded-md `,
              `${transitionBgClass}`,
            ].join("");
            const linkClass = `font-normal w-full py-3 px-7 ${centerOpenClass}`;

            return (
              <li
                key={label}
                className={menuItemClass}
                onClick={handleClickSidebarItem}>
                <Link className={linkClass} href={href}>
                  <span className="w-5">{content}</span>
                  <span className={`${hiddenOpenClass} pl-2`}>{label}</span>
                </Link>
              </li>
            );
          })}
        </ul>
      </div>
    </>
  );
};

export default SideBar;
