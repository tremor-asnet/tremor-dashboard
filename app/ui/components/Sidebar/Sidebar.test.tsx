import { fireEvent, render } from "@testing-library/react";

// Constants
import { ROUTES } from "@/constants";

// Component
import Sidebar from "./Sidebar";

describe("Sidebar component", () => {
  const mockFunction = jest.fn();

  const propsDefault = {
    avatarUrl: "/images/avatar/avatar-sm.webp",
    name: "Brooklyn Alice",
    pathname: ROUTES.PROJECTS,
    isCollapse: true,
    toggleSidebar: mockFunction,
    onSignOut: mockFunction,
  };

  const sidebar = () => {
    return render(<Sidebar {...propsDefault} />);
  };

  it("Should render Sidebar snapshot correctly", () => {
    expect(sidebar()).toMatchSnapshot();
  });

  it("should call toggleSidebar when clicked outside", () => {
    const toggleSidebar = jest.fn();
    const onSignOut = jest.fn();
    const { container } = render(
      <div>
        <div id="outside-element">Outside Element</div>
        <Sidebar
          avatarUrl=""
          name=""
          isCollapse={true}
          toggleSidebar={toggleSidebar}
          pathname=""
          onSignOut={onSignOut}
        />
      </div>,
    );

    const outsideElement = container.querySelector("#outside-element");
    fireEvent.mouseDown(outsideElement!);

    expect(toggleSidebar).toHaveBeenCalledTimes(1);
  });

  // Mock window.innerWidth
  const originalInnerWidth = window.innerWidth;
  beforeAll(() => {
    Object.defineProperty(window, "innerWidth", {
      writable: true,
      configurable: true,
      value: 800, // Set a value greater than BREAKPOINTS === 1280
    });
  });

  it("closes sidebar when window width is smaller than or equal to BREAKPOINTS === 1280", () => {
    const toggleSidebar = jest.fn();

    const { container } = render(
      <Sidebar
        avatarUrl=""
        name=""
        isCollapse={true}
        toggleSidebar={toggleSidebar}
        pathname=""
        onSignOut={() => Promise.resolve()}
      />,
    );

    const sidebar = container.querySelector(".sidebar");
    fireEvent.click(sidebar!);

    expect(toggleSidebar).toHaveBeenCalledTimes(0);
  });

  // Restore window.innerWidth
  afterAll(() => {
    Object.defineProperty(window, "innerWidth", {
      writable: true,
      configurable: true,
      value: originalInnerWidth,
    });
  });
});
