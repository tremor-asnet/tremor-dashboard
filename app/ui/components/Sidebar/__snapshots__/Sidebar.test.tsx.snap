// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component Should render Sidebar snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sidebar antialiased shadow-box-sidebar bg-gradient-primary dark:bg-none dark:bg-dark-gradient-primary w-[250px] rounded-xl z-50 px-4 pt-6 overflow-y-auto overflow-x-hidden fixed top-4 left-4 h-[calc(100vh-2rem)] transition-all ease-in translate-x-0 xl:w-[100px] delay-10 duration-300"
      >
        <div
          class="tremor-Flex-root flex w-full flex-row justify-end items-center"
        >
          <div
            class="border p-1 rounded-s-md -mr-4 bg-white cursor-pointer"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 256 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
              />
            </svg>
          </div>
        </div>
        <a
          href="/analytics"
        >
          <div
            class="tremor-Flex-root flex w-full flex-row items-center justify-normal pl-5 gap-1 pb-2 flex-nowrap relative"
          >
            <img
              alt="Tremor Dashboard"
              data-nimg="1"
              decoding="async"
              height="28"
              loading="lazy"
              src="http://localhost/_next/image?url=%2Fassets%2Fimages%2Fsidebar-logo.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fassets%2Fimages%2Fsidebar-logo.webp&w=32&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fsidebar-logo.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="28"
            />
            <p
              class="font-semibold dark:text-dark-tremor-content-strong text-white text-tremor-default min-w-[200px] flex flex-nowrap xl:hidden delay-300 duration-500"
            >
              Tremor Dashboard PRO
            </p>
          </div>
        </a>
        <div
          class="h-px bg-gradient-divider my-4 opacity-25"
        />
        <div
          class="tremor-AccordionList-root rounded-tremor-default shadow-tremor-card dark:shadow-dark-tremor-card bg-transparent shadow-none border-0"
        >
          <div
            class="tremor-Accordion-root overflow-hidden border-tremor-border dark:border-dark-tremor-border rounded-t-tremor-default bg-inherit dark:bg-none dark:bg-dark-gradient-primary border-0 rounded-md"
            data-headlessui-state=""
          >
            <button
              aria-expanded="false"
              class="tremor-AccordionHeader-root w-full items-center justify-between px-4 dark:text-dark-tremor-content-emphasis min-w-[210px] flex text-[rgba(255,255,255,0.5)] py-2 xl:w-[200%]"
              data-headlessui-state=""
              id="headlessui-disclosure-button-:r0:"
              type="button"
            >
              <div
                class="tremor-AccordionHeader-children flex flex-1 text-inherit mr-4"
              >
                <div
                  class="inline-flex rounded-full overflow-hidden flex items-center justify-center"
                >
                  <img
                    alt="Avatar small"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="36"
                    sizes="(max-width: 768px) 100vw, 33vw"
                    src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=3840&q=75 3840w"
                    style="color: transparent;"
                    width="36"
                  />
                </div>
                <p
                  class="text-tremor-default leading-0 dark:text-dark-primary min-w-[100px] flex flex-nowrap text-white self-center ml-[7px] xl:hidden"
                >
                  Brooklyn Alice
                </p>
              </div>
              <div>
                <svg
                  class="tremor-AccordionHeader-arrowIcon h-5 w-5 -mr-1 text-tremor-content-subtle dark:text-dark-tremor-content-subtle transition-all -rotate-180"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.9999 10.8284L7.0502 15.7782L5.63599 14.364L11.9999 8L18.3639 14.364L16.9497 15.7782L11.9999 10.8284Z"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          class="h-px bg-gradient-divider my-4 opacity-25"
        />
        <ul>
          <li
            class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
          >
            <a
              class="font-normal w-full py-3 px-7 xl:justify-center"
              href="/analytics"
            >
              <span
                class="w-5"
              >
                AN
              </span>
              <span
                class="xl:hidden pl-2"
              >
                Analytics
              </span>
            </a>
          </li>
          <li
            class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
          >
            <a
              class="font-normal w-full py-3 px-7 xl:justify-center"
              href="/sales"
            >
              <span
                class="w-5"
              >
                S
              </span>
              <span
                class="xl:hidden pl-2"
              >
                Sales
              </span>
            </a>
          </li>
          <li
            class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center bg-primary hover:!bg-primary !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
          >
            <a
              class="font-normal w-full py-3 px-7 xl:justify-center"
              href="/all-projects"
            >
              <span
                class="w-5"
              >
                A
              </span>
              <span
                class="xl:hidden pl-2"
              >
                All Projects
              </span>
            </a>
          </li>
          <li
            class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
          >
            <a
              class="font-normal w-full py-3 px-7 xl:justify-center"
              href="/order-list"
            >
              <span
                class="w-5"
              >
                O
              </span>
              <span
                class="xl:hidden pl-2"
              >
                Order List
              </span>
            </a>
          </li>
          <li
            class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
          >
            <a
              class="font-normal w-full py-3 px-7 xl:justify-center"
              href="/product-list"
            >
              <span
                class="w-5"
              >
                P
              </span>
              <span
                class="xl:hidden pl-2"
              >
                Product List
              </span>
            </a>
          </li>
          <li
            class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
          >
            <a
              class="font-normal w-full py-3 px-7 xl:justify-center"
              href="/billing"
            >
              <span
                class="w-5"
              >
                B
              </span>
              <span
                class="xl:hidden pl-2"
              >
                Billing
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sidebar antialiased shadow-box-sidebar bg-gradient-primary dark:bg-none dark:bg-dark-gradient-primary w-[250px] rounded-xl z-50 px-4 pt-6 overflow-y-auto overflow-x-hidden fixed top-4 left-4 h-[calc(100vh-2rem)] transition-all ease-in translate-x-0 xl:w-[100px] delay-10 duration-300"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-end items-center"
      >
        <div
          class="border p-1 rounded-s-md -mr-4 bg-white cursor-pointer"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 256 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            />
          </svg>
        </div>
      </div>
      <a
        href="/analytics"
      >
        <div
          class="tremor-Flex-root flex w-full flex-row items-center justify-normal pl-5 gap-1 pb-2 flex-nowrap relative"
        >
          <img
            alt="Tremor Dashboard"
            data-nimg="1"
            decoding="async"
            height="28"
            loading="lazy"
            src="http://localhost/_next/image?url=%2Fassets%2Fimages%2Fsidebar-logo.webp&w=64&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fsidebar-logo.webp&w=32&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fsidebar-logo.webp&w=64&q=75 2x"
            style="color: transparent;"
            width="28"
          />
          <p
            class="font-semibold dark:text-dark-tremor-content-strong text-white text-tremor-default min-w-[200px] flex flex-nowrap xl:hidden delay-300 duration-500"
          >
            Tremor Dashboard PRO
          </p>
        </div>
      </a>
      <div
        class="h-px bg-gradient-divider my-4 opacity-25"
      />
      <div
        class="tremor-AccordionList-root rounded-tremor-default shadow-tremor-card dark:shadow-dark-tremor-card bg-transparent shadow-none border-0"
      >
        <div
          class="tremor-Accordion-root overflow-hidden border-tremor-border dark:border-dark-tremor-border rounded-t-tremor-default bg-inherit dark:bg-none dark:bg-dark-gradient-primary border-0 rounded-md"
          data-headlessui-state=""
        >
          <button
            aria-expanded="false"
            class="tremor-AccordionHeader-root w-full items-center justify-between px-4 dark:text-dark-tremor-content-emphasis min-w-[210px] flex text-[rgba(255,255,255,0.5)] py-2 xl:w-[200%]"
            data-headlessui-state=""
            id="headlessui-disclosure-button-:r0:"
            type="button"
          >
            <div
              class="tremor-AccordionHeader-children flex flex-1 text-inherit mr-4"
            >
              <div
                class="inline-flex rounded-full overflow-hidden flex items-center justify-center"
              >
                <img
                  alt="Avatar small"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="36"
                  sizes="(max-width: 768px) 100vw, 33vw"
                  src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=3840&q=75"
                  srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-sm.webp&w=3840&q=75 3840w"
                  style="color: transparent;"
                  width="36"
                />
              </div>
              <p
                class="text-tremor-default leading-0 dark:text-dark-primary min-w-[100px] flex flex-nowrap text-white self-center ml-[7px] xl:hidden"
              >
                Brooklyn Alice
              </p>
            </div>
            <div>
              <svg
                class="tremor-AccordionHeader-arrowIcon h-5 w-5 -mr-1 text-tremor-content-subtle dark:text-dark-tremor-content-subtle transition-all -rotate-180"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9999 10.8284L7.0502 15.7782L5.63599 14.364L11.9999 8L18.3639 14.364L16.9497 15.7782L11.9999 10.8284Z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
      <div
        class="h-px bg-gradient-divider my-4 opacity-25"
      />
      <ul>
        <li
          class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
        >
          <a
            class="font-normal w-full py-3 px-7 xl:justify-center"
            href="/analytics"
          >
            <span
              class="w-5"
            >
              AN
            </span>
            <span
              class="xl:hidden pl-2"
            >
              Analytics
            </span>
          </a>
        </li>
        <li
          class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
        >
          <a
            class="font-normal w-full py-3 px-7 xl:justify-center"
            href="/sales"
          >
            <span
              class="w-5"
            >
              S
            </span>
            <span
              class="xl:hidden pl-2"
            >
              Sales
            </span>
          </a>
        </li>
        <li
          class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center bg-primary hover:!bg-primary !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
        >
          <a
            class="font-normal w-full py-3 px-7 xl:justify-center"
            href="/all-projects"
          >
            <span
              class="w-5"
            >
              A
            </span>
            <span
              class="xl:hidden pl-2"
            >
              All Projects
            </span>
          </a>
        </li>
        <li
          class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
        >
          <a
            class="font-normal w-full py-3 px-7 xl:justify-center"
            href="/order-list"
          >
            <span
              class="w-5"
            >
              O
            </span>
            <span
              class="xl:hidden pl-2"
            >
              Order List
            </span>
          </a>
        </li>
        <li
          class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
        >
          <a
            class="font-normal w-full py-3 px-7 xl:justify-center"
            href="/product-list"
          >
            <span
              class="w-5"
            >
              P
            </span>
            <span
              class="xl:hidden pl-2"
            >
              Product List
            </span>
          </a>
        </li>
        <li
          class="tremor-ListItem-root w-full flex justify-between items-center truncate text-tremor-default !p-0 leading-[26px] rounded-md my-[3px] text-center hover:bg-none !p-0 leading-[26px] mt-1 rounded-md transition-[background-color] duration-300 ease-[cubic-bezier(0.4,0,0.6,1)] delay-20"
        >
          <a
            class="font-normal w-full py-3 px-7 xl:justify-center"
            href="/billing"
          >
            <span
              class="w-5"
            >
              B
            </span>
            <span
              class="xl:hidden pl-2"
            >
              Billing
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
